### PR TITLE
Piano di qualifica - 48

### LINKED ISSUE #48: Piano di qualifica - qualità di prodotto
### Oggetto di discussione

VI_2024-11-22

### Codice Verbale

VI 7.6

### Link Verbale

/.

### Indica il ruolo

Analista

### Informazioni minime da implementare

- [ ] Stesura sezione "qualità di prodotto"

### PR DESCRIPTION
**Pull request by EliaBergamin**
- Aggiunta sezione Qualità di prodotto
- riviste sezioni precedenti
- close #48

### CODE REVIEW

File: Docs/Generali/piano_di_qualifica/02_qual_di_proc.tex [hunk 1]
```diff
@@ -64,11 +62,11 @@ \subsection{Metriche}
...
-    MPC11 & Tempo di produzione & $\leq 2$h & $\leq 1$h\tabularnewline
-    MPC12 & Metriche soddisfatte & $\geq 90$\% & 100\%\tabularnewline
-    MPC13 & Code Coverage (CC) & $\geq 75$\% & $\geq 90$\%\tabularnewline
-    MPC14 & Passed Test Cases Percentage (PTCP) & $\geq 90$\% & 100\%\tabularnewline
+    MPC10 & Completezza dei documenti & 100\% & 100\%\tabularnewline
+    MPC11 & Metriche soddisfatte & $\geq 80$\% & 100\%\tabularnewline
+    MPC12 & Code Coverage (CC) & $\geq 75$\% & $\geq 90$\%\tabularnewline
+    MPC13 & Statement Coverage (CC) & $\geq 80$\% & 100\%\tabularnewline
+    MPC14 & Branch Coverage (CC) & $\geq 80$\% & 100\%\tabularnewline
     MPC15 & Rischi non previsti & $\leq 5$ & 0\tabularnewline
```
Comment (ElenaaaCh) [inline Docs/Generali/piano_di_qualifica/02_qual_di_proc.tex:71]: Per maggior coerenza scriverei tutto in inglese

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+\prodQualityTable{
+
+    Funzionalità & 
+     Capacità di un prodotto software di fornire le funzioni, espresse ed implicite, necessarie 
+     per operare in un determinato contesto & 
+     MPD3, MPD4, MPD5\tabularnewline
+    Affidabilità & 
+     Capacità di un prodotto software di mantenere il livello di prestazione 
+     quando viene utilizzato in condizioni specificate &
```
Comment (ElenaaaCh) [inline Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:30]: definizione by Tullio : capacità di svolgere correttamente il suo compito quando utilizzato

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+     MPD6, MPD7, MPD8\tabularnewline
+    Usabilità & 
+     Capacità di un prodotto software di essere compreso e appreso & 
+     MPD9\tabularnewline
+    Efficienza & 
+     Capacità di un prodotto software di realizzare le funzioni richieste nel 
+     minor tempo possibile ed utilizzando nel miglior modo le risorse necessarie &
```
Comment (ElenaaaCh) [inline Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:37]: aggiungi: solo le risorse strettamente necessarie

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+     MPD10, MPD11\tabularnewline
+    Manutenibilità & 
+     Capacità di un prodotto software di essere modificato &
```
Comment (ElenaaaCh) [inline Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:40]: aggiungi: modificato nel tempo per rispondere a nuove esigenze

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+     MPD9\tabularnewline
+    Efficienza & 
+     Capacità di un prodotto software di realizzare le funzioni richieste nel 
+     minor tempo possibile ed utilizzando nel miglior modo le risorse necessarie &
+     MPD10, MPD11\tabularnewline
+    Manutenibilità & 
+     Capacità di un prodotto software di essere modificato & 
+     MPD12, MPD13, MPD14, MPD15, MPD16\tabularnewline
+    Portabilità & 
+     Capacità di un prodotto software di poter essere trasportato da un ambiente ad un altro &
```
Comment (ElenaaaCh) [inline Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:43]: ambiente operativo

**Review by ElenaaaCh — CHANGES_REQUESTED** [review]
Sistemare le frasi indicate

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex (hunk 2 above)
Comment (EliaBergamin) [reply Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:30]: questa è la definizione dello standard (anche le altre)

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex (hunk 3 above)
Comment (EliaBergamin) [reply Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:37]: https://cdn.standards.iteh.ai/samples/35755/21d54ab4340b47acb715a549116fbc33/ISO-IEC-25030-2007.pdf
(definizioni riprese dallo standard 9126)
5.4 Software quality model
The quality of a system is the result of the quality of the system elements and their interaction. This
International Standard focuses on the quality of the software as part of a system.
Software quality is the capability of the software product to satisfy stated and implied needs when
used under specified conditions. The software product quality model provided in ISO/IEC 9126-1
[ISO/IEC 25010] defines six quality characteristics:
• Functionality: The capability of the software product to provide functions which meet stated
and implied needs when the software is used under specified conditions.
• Reliability: The capability of the software product to maintain a specified level of performance when used under specified conditions
• Usability: The capability of the software product to be understood, learned, used and attractive to the user, when used under specified conditions.
• Efficiency: The capability of the software product to provide appropriate performance, relative to the amount of resources used, under stated conditions.
• Maintainability: The capability of the software product to be modified. Modifications may
include corrections, improvements or adaptation of the software to changes in environment,
and in requirements and functional specifications.
• Portability: The capability of the software product to be transferred from one environment to
another.

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex (hunk 4 above)
Comment (EliaBergamin) [reply Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:40]: vedi sopra

File: Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex (hunk 5 above)
Comment (EliaBergamin) [reply Docs/Generali/piano_di_qualifica/03_qual_di_prod.tex:43]: vedi sopra

**Review by ElenaaaCh — APPROVED** [review]

**Pull request merged**